core/tests/puppeteer-acceptance-tests/spec/exploration-editor-tests/exploration-publisher-drafter.spec.js

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Acceptance Test for Exploration Creator and Exploration Manager
+ */
+
+const userFactory = require(
+  '../../puppeteer-testing-utilities/user-factory.js');
+const testConstants = require(
+  '../../puppeteer-testing-utilities/test-constants.js');
+
+const DEFAULT_SPEC_TIMEOUT = testConstants.DEFAULT_SPEC_TIMEOUT;
+
+describe('Exploration Publisher, Saver and Drafter', function() {
+  let explorationCreator = null;
+  beforeAll(async function() {
+    explorationCreator = await userFactory.createNewExplorationCreator(
+      'explorationAdm', 'exploration_creator@example.com');
+    explorationVisitor = await userFactory.createNewExplorationCreator(
+      'explorationLearner', 'exploration_visitor@example.com');
+  }, DEFAULT_SPEC_TIMEOUT);
+
+  it('should perform exploration management actions',
+    async function() {
+      await explorationCreator.goToDashboardUrl();
+      await explorationCreator.goToEditorSection();
+      await explorationCreator.updateCardName('Test question');
+      await explorationCreator.updateExplorationIntroText(
+        'Exploration intro text');
+      await explorationCreator.addInteraction();
+      await explorationCreator.explorationCreatedSuccessfully();
+
+      await explorationCreator.goToSettingsTab();
+      await explorationCreator.updateTitle('Your Title Here');
+      await explorationCreator.updateGoal('NeedSuccessInLifeAndMoney');
+      await explorationCreator.selectCategory();
+      await explorationCreator.selectLanguage();
+      await explorationCreator.addTags('Your Tag Here');
+      await explorationCreator.successfullyUpdatedSettings();
+
+      await explorationCreator.publishExploration();
+
+      await explorationCreator.addSomeChanges();
```

Comment:
This seems too vague. If you want to check that changes are discarded then you need to be more specific here so that it's clear to the reader of this file what the state before / after the changes is. For example:

```
explorationCreator.updateTitle('Old Title')
...
explorationCreator.updateTitle('New Title')
explorationCreator.expectTitleToBe('New Title')

explorationCreator.discardChanges()
explorationCreator.expectTitleToBe('Old Title')
```

(This is just an example, you can do whatever update you like, but notice how the structure here makes it easy to see that the action of discarding changes resulted in a revert to the old title. This is not clear at all from the code here.)

Reply:
@seanlip 
<img width="695" alt="Screenshot 2024-03-03 at 2 00 26 AM" src="https://github.com/oppia/oppia/assets/136263179/0aee77df-3319-4b6f-aba4-fde4aadb00f5">

Note : Here in last line we have `Old TitleNew Title` because we are not erasing old title just modifying.
Is it look clear now as per -> 
<img width="328" alt="Screenshot 2024-03-03 at 2 01 30 AM" src="https://github.com/oppia/oppia/assets/136263179/ad287020-4901-48fb-b69d-f298e12c9b15">

Reply:
The flow seems fine but I don't think people will take "updateTitle" to mean "add characters at the end", they will take it to mean "replace the value with X". Please fix accordingly.

Reply:
changed to addTitle() PTAL .